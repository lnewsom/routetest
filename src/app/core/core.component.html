<app-header [coreId]="this.coreId"></app-header>
<router-outlet></router-outlet>
