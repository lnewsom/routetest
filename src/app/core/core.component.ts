import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';

@Component({
  selector: 'app-core',
  templateUrl: './core.component.html',
  styleUrls: ['./core.component.scss']
})
export class CoreComponent implements OnInit {

  constructor(private route: ActivatedRoute) { }

  ngOnInit() {
    this.route.parent.params.subscribe((params) => {
      console.log('Core Params: ' , params);
    });
  }

}
