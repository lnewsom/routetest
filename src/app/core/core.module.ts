import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CoreComponent } from './core.component';
import { HeaderComponent } from './header/header.component';
import { AppRoutingModule } from '../app-routing.module';


@NgModule({
  declarations: [
    CoreComponent,
    HeaderComponent
  ],
  imports: [
    CommonModule,
    AppRoutingModule
    ], exports: [
    HeaderComponent
  ]
})
export class CoreModule { }
