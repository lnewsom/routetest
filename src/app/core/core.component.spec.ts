import { Chance } from 'chance';
import { CoreComponent } from './core.component';
import { ActivatedRoute } from '@angular/router';
import { of } from 'rxjs';
import { async, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

const chance: Chance.Chance = new Chance();
jest.mock('@angular/router');

describe('CoreComponent', () => {
  const expectedCoreId: string = chance.string();
  let mockedActivatedRoute: ActivatedRoute;
  let underTest: CoreComponent;

  describe('ngOnInit', () => {
    beforeEach(() => {
      mockedActivatedRoute = new ActivatedRoute();
      mockedActivatedRoute.params = of({coreId: expectedCoreId});
      underTest = new CoreComponent(mockedActivatedRoute);
      underTest.ngOnInit();
    });

    test('sets the correct coreId', () => {
      expect(underTest.coreId).toEqual(expectedCoreId);
    });
  });

  describe('snapshot', () => {
    beforeEach(async(() => {
      TestBed.configureTestingModule({
        providers: [ActivatedRoute],
        declarations: [CoreComponent],
        schemas: [NO_ERRORS_SCHEMA]
      }).compileComponents();
    }));

    test('snapshot matches', () => {
      const fixture = TestBed.createComponent(CoreComponent);
      expect(fixture).toMatchSnapshot();
    });
  });

});
