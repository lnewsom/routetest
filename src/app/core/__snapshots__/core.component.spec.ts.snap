// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreComponent snapshot snapshot matches 1`] = `
<app-core
  route={[Function ActivatedRoute]}
>
  <app-header /><router-outlet />
</app-core>
`;
